src/android/orb.yml

```diff
@@ -121,7 +144,7 @@ commands:
           command: gcloud auth activate-service-account --key-file "<<parameters.key-file>>"
       - run:
           name: Run tests on Firebase Test Lab
-          no_output_timeout: 1h # Set to a large value as firebase manages its own timeout.
+          no_output_timeout: 3h # Set to a large value as firebase manages its own timeout.
```

Comment:
Do we really need to set it this high? Does Firebase returns an error on its own timeout, or the CI will be stuck for 3 hours in that case?

Reply:
Firebase has its own timeout and will return an error on such timeout (see the `timeout` parameter of the orb's command)

That's actually something I'm currently trying to test, but I struggle a bit to check it because I can only test with 2x2 matrices, while the time taken for a 2x21 matrix (the target for WPAndroid screenshots) might take longer depending on if Firebase limit parallelism on bigger matrices (e.g. maybe it only allows a maximum of 4 parallel runs in larger matrices, implementing some queue mechanism internally to throttle the rest of the runs and also depending on the physical devices they actually have available at a given time?). If they have such a queuing/throttling mechanism for larger matrix sizes, we might need the larger timeout on the CircleCI job itself in case the overall matrix of 42 devices takes a long time. Since running the screenshots with `screengrab` locally on my machine for 2x21 devices took ≈ 100 minutes on my Mac, I figured that 1h limit could be too low even on FTL, but depending on their parallelism policy, maybe not.

Now that my quota for today has reset, I will soon be able to launch a new build to look at this more in details, especially look at the stdout of the CircleCI step while the matrix is executing on FTL, will report back.

Reply:
Ok, seems that the `gcloud` command is [reporting back regularly by printing stuff on `stdout` during the progress of the FTL matrix execution](https://app.circleci.com/pipelines/github/AliSoftware/WordPress-Android/109/workflows/3a18769e-1f73-4c59-9a9c-a4f3f2dafc44/jobs/290).

So maybe we don't need to increase CircleCI's own `no_output_timeout` here after all

My WPAndroid matrix execution failed for a different timeout reason ([JUnit assertion timeout while waiting for the Post screen it seems](https://console.firebase.google.com/project/wpandroid-74b6c/testlab/histories/bh.1311dc7698e15a83/matrices/6647941184985002595/executions/bs.bd69ab0127b4a360/testcases/1/test-cases) – which is the first time that I see it happen, so no idea why I suddenly get that while same tests were working yesterday and tests are working locally…), so I just burned my only execution of the day and reached the quota and won't be able to test again, but I think it's reasonable to revert the change here to 1h anyway.